### DISCUSSION TITLE
Material UI Select and setValue method

### DISCUSSION THREAD

**Discussion by sigsiroos** [react-hook-form/react-hook-form#3344] [Q&A]
I'm trying to use `setValue` to change the value of my `Select` from `Material UI` but it doesn't seem to work even though the value seems to be set. Why is this?

The reason I want to achieve this is that in my project where I use graphql, after a mutation I want to set my `Select` value with the result from the mutation

Sandbox: https://codesandbox.io/s/somaterialselecthookform-forked-hbuyv?file=/src/App.js

**Comment by bluebill1049** [comment]
if you want to use `setValue` to update your select, you need to wrapper your external controlled component with Controller first. You shouldn't need to use setValue internally.

eg:

```
<Controller name="test" render={props => <Swtich {...props} />} />

<button onClick={() => setValue('test', 'value')}>Test</button>
```

**Reply by sigsiroos** [reply]
Yes that's what I was thinking first as well so that value get's properly passed to the `Select` if I've understood it correctly?
But thought I tried it and still didn't work, anyways tried editing the sandbox to test this but got the following error message

```
Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.

Check the render method of `Controller`.
```
https://codesandbox.io/s/somaterialselecthookform-forked-rd6o8?file=/src/App.js

Do you have time to help me out since I can't understand why this happens with render prop? Seems to work on a different sandbox that you have in the RHF docs but on mine it doesn't
